issue solved, but need to update guide - [mumble] broken, no package downloads available with release 1.4.230 and beyond

With the [latest release of mumble server](https://github.com/mumble-voip/mumble/releases/tag/v1.4.230) (murmur), the minor version increased to **v1.4**. With that, the `murmur-static_x86.tar.bz2` file (linux server) is no longer available in the release assets in github.

So the download instructions from the guide will no longer work by copy & paste:
```
wget https://github.com/mumble-voip/mumble/releases/download/66.6.6/murmur-static_x86-66.6.6.tar.bz2
```

The Mumble download site does also not provide official packages of linux mumble server. Instead it states:
> Your distribution probably provides official packages for Mumble. Please refer to your OS packages. Distributions with official packages include Debian, Ubuntu, Fedora, openSUSE, Arch Linux, Mandriva/ROSA/Unity.

Nothing mentioned about CentOS.

---

The **previous release 1.3.4** also contains the following phrase in its release notes:
> **Warning** ⚠️ 
>
> The static server binary for Linux is using an outdated version of OpenSSL - see [4001](https://github.com/mumble-voip/mumble/issues/4001) for details. This warning only applies to the package released by us named murmur-static_x86. It does not apply to our Ubuntu PPA releases or other maintained packages.

[tl;dr of mumble issue no. 4001](https://github.com/mumble-voip/mumble/issues/4001#issuecomment-927142091): **won't fix!** ⚠️ 
> We will most likely not provide new builds for the 1.3.x series though. We are preparing to release 1.4.x that does not have this issue in the first place so that should cover it.

So people should probably update any mumble-server instance to 1.4.x if it was installed following the instructions from the guide.

I'm stopping my investigations here. @Stunkymonkey, as you initially wrote the mumble guide, do you have time to check on this (and eventually update the guide)?

I just gave it a try, but linking seems to fail for now:

```
[ 98%] Building CXX object src/murmur/CMakeFiles/mumble-server.dir/Murmur.cpp.o
[100%] Linking CXX executable ../../mumble-server
/opt/rh/devtoolset-10/root/usr/libexec/gcc/x86_64-redhat-linux/10/ld: CMakeFiles/mumble-server.dir/mumble-server_plugin_import.cpp.o: in function `_GLOBAL__sub_I_mumble_server_plugin_import.cpp':
mumble-server_plugin_import.cpp:(.text.startup+0x5): undefined reference to `qt_static_plugin_QSQLiteDriverPlugin()'
collect2: error: ld returned 1 exit status
make[2]: *** [mumble-server] Error 1
make[1]: *** [src/murmur/CMakeFiles/mumble-server.dir/all] Error 2
make: *** [all] Error 2
Error: Process completed with exit code 2.
```

If someone has an idea about this, I am open for it. I have been using GitHub Actions for building, so the build script is available at https://github.com/FriedrichFroebel/murmur_tests/. By the way: The build seems to take about 6 minutes there.

> undefined reference to `qt_static_plugin_QSQLiteDriverPlugin()'

On Ubuntu I needed to install the vcpkg-package libmariadb (and uninstall libmysql as they don't go together) ... might that help with your error?

Sadly (if I don't bother about the vcpkg dependency errors) my build process fails already at the beginning on `~/mumble/src/MumbleUDP.proto` and `~/mumble/src/Mumble.proto`, because it needs proto3 (from package protobuf), but i have only proto2 and have not found a way to install protbuf v3 for now...
So as @franok mentioned, there are vast problems with dependencies in CentOS 7 :disappointed:

--- a/source/guide_mumble.rst
+++ b/source/guide_mumble.rst
@@ -1,4 +1,4 @@
-.. author:: stunkymonkey <http://stunkymonkey.de>
+.. author:: stunkymonkey <http://stunkymonkey.de>, updated by franok <https://franok.de>
 .. highlight:: console
 
 .. tag:: lang-cpp
@@ -16,7 +16,7 @@ Mumble
 
 .. tag_list::
 
-Mumble_ is a voice over IP (VoIP) application primarily designed for use by gamers and is similar to programs such as TeamSpeak_.
+Mumble_ is a voice over IP (VoIP) application primarily designed for use by gamers and is similar to programs such as TeamSpeak_. In contrast to Teamspeak, Mumble is an open-source project.
 
 Mumble uses a client–server architecture which allows users to talk to each other via the same server (murmur). It has a very simple administrative interface and features high sound quality and low latency. All communication is encrypted to ensure user privacy.
 
@@ -24,7 +24,6 @@ Mumble uses a client–server architecture which allows users to talk to each ot
 
 .. note:: For this guide you should be familiar with the basic concepts of
 
-  * MySQL_
   * supervisord_
 
 License
@@ -34,67 +33,61 @@ All relevant legal information can be found here
 
   * https://github.com/mumble-voip/mumble/blob/master/LICENSE
 
-Prerequisites
+Pre-Note
 =============
 
-.. include:: includes/my-print-defaults.rst
+.. note::
+  The official Mumble project decided to no longer provide static murmur server binaries starting with version 1.4.x.
 
-You need a database for mumble:
+  `Some fellow Ubernauts have undertaken efforts <https://github.com/Uberspace/lab/issues/1191>`_ to compile a static murmur binary on their own. You can have a look at it on GitHub_.
 
-.. code-block:: console
+  This guide uses the self-compiled server binary, which currently comes with some limitations, i.e. MySQL is currently not supported as a database. For Ubernauts who have been hosting a Mumble server based on this guide before Mumble v1.4, this might be a breaking change. The default database for a Mumble server is SQLite and requires no further configuration.
 
-  [isabell@stardust ~]$ mysql -e "CREATE  DATABASE ${USER}_mumble"
-  [isabell@stardust ~]$
+  Please read the `release notes <https://github.com/franok/mumble-build-container/releases>`_ of every new release. Releases of the self-compiled server binary may occur more often than the offical Mumble releases. However, no warranty is made regarding up-to-dateness and functional correctness. Consider subscribing to new releases, e.g. by using :lab:`uu-notify <guide_uu-notify>`.
+
 
 Installation
 ============
 
-Check the Mumble_ website or GitHub_ for the feed_ and copy the download link to the ``murmur-static``-file. Replace the URL with the one you just copied.
+Check the `GitHub release section`_ and copy the release tag version of the latest release. Set the variable ``MURMUR_VERSION`` to the version you just copied.
 
-.. note:: the server for mumble is called murmur
+.. note:: The server for mumble is called murmur
 
 .. code-block:: console
 
-  [isabell@stardust ~]$ wget https://github.com/mumble-voip/mumble/releases/download/66.6.6/murmur-static_x86-66.6.6.tar.bz2
-  --2018-10-06 21:14:34--  https://github.com/mumble-voip/mumble/releases/download/66.6.6/murmur-static_x86-66.6.6.tar.bz2
+  [isabell@stardust ~]$ mkdir mumble
+  [isabell@stardust ~]$ cd mumble/
+  [isabell@stardust mumble]$ MURMUR_VERSION=v0.0.0_build-YYYY-000
+  [isabell@stardust mumble]$ wget https://github.com/franok/mumble-build-container/releases/download/$MURMUR_VERSION/murmur-static.zip
+  --2022-12-14 20:01:29--  https://github.com/franok/mumble-build-container/releases/download/v1.4.287/murmur-static.zip
   Resolving github.com (github.com)... 192.30.253.113, 192.30.253.112
   Connecting to github.com (github.com)|192.30.253.113|:443... connected.
   HTTP request sent, awaiting response... 302 Found
-  Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/1413319/a7d2b00a-e4a8-11e6-80bb-d224f64df619?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20181006%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20181006T191435Z&X-Amz-Expires=300&X-Amz-Signature=b2af6f60968f675bb8427af7341716d75a1ee162f822abdd878877514b9205be&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dmurmur-static_x86-66.6.6.tar.bz2&response-content-type=application%2Foctet-stream [following]
-  --2018-10-06 21:14:35--  https://github-production-release-asset-2e65be.s3.amazonaws.com/1413319/a7d2b00a-e4a8-11e6-80bb-d224f64df619?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20181006%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20181006T191435Z&X-Amz-Expires=300&X-Amz-Signature=b2af6f60968f675bb8427af7341716d75a1ee162f822abdd878877514b9205be&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dmurmur-static_x86-66.6.6.tar.bz2&response-content-type=application%2Foctet-stream
-  Resolving github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)... 52.216.2.32
-  Connecting to github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)|52.216.2.32|:443... connected.
-  HTTP request sent, awaiting response... 200 OK
-  Length: 11939824 (11M) [application/octet-stream]
-  Saving to: ‘murmur-static_x86-66.6.6.tar.bz2’
+  Location: ...
+  Saving to: ‘murmur-static.zip’
 
-  100%[======================================================================================================================================>] 11,939,824  7.45MB/s   in 1.5s
+  murmur-static.zip     100%[=============================>]   7,39M  4,47MB/s    in 1,7s
 
-  2018-10-06 21:14:37 (7.45 MB/s) - ‘murmur-static_x86-66.6.6.tar.bz2’ saved [11939824/11939824]
+  2022-12-14 20:01:33 (4,47 MB/s) - ‘murmur-static.zip’ saved [7753035/7753035]
 
-  [isabell@stardust ~]$
+  [isabell@stardust mumble]$
 
 
-extract it and rename it to mumble:
+Extract the zip file:
 
 
 .. code-block:: console
 
-  [isabell@stardust ~]$ tar xvjf murmur-static_x86-66.6.6.tar.bz2
-  murmur-static_x86-66.6.6/
-  murmur-static_x86-66.6.6/murmur.x86
-  murmur-static_x86-66.6.6/LICENSE
-  murmur-static_x86-66.6.6/dbus/
-  murmur-static_x86-66.6.6/dbus/weblist.pl
-  murmur-static_x86-66.6.6/dbus/murmur.pl
-  murmur-static_x86-66.6.6/ice/
-  murmur-static_x86-66.6.6/ice/icedemo.php
-  murmur-static_x86-66.6.6/ice/Murmur.ice
-  murmur-static_x86-66.6.6/ice/weblist.php
-  murmur-static_x86-66.6.6/murmur.ini
-  murmur-static_x86-66.6.6/README
-  [isabell@stardust ~]$ mv murmur-static_x86-66.6.6/ mumble/
-  [isabell@stardust ~]$
+  [isabell@stardust mumble]$ unzip -l murmur-static.zip
+  Archive:  murmur-static.zip
+    Length      Date    Time    Name
+  ---------  ---------- -----   ----
+      18545  2022-12-06 21:27   murmur.ini
+   18621464  2022-12-06 21:28   murmur.x86_64
+  ---------                     -------
+  18640009                     2 files
+
+  [isabell@stardust mumble]$
 
 
 Configuration
@@ -105,21 +98,17 @@ Configure port
 
 .. include:: includes/open-port.rst
 
-edit config file ``~/mumble/murmur.ini`` to specify the desired port, domain and mysql-database. Minimal config would look like this:
+Edit the config file ``~/mumble/murmur.ini`` to specify the desired port. Minimal config would look like this:
 
 .. code-block:: ini
-  :emphasize-lines: 1,3,4,7
+  :emphasize-lines: 2,3
 
-  database=<username>_mumble
-  dbDriver=QMYSQL
-  dbUsername=<username>
-  dbPassword=<mysql_password>
-  dbOpts="UNIX_SOCKET=/var/lib/mysql/mysql.sock"
   welcometext="<br />Welcome to this server running <b>Murmur</b>.<br />Enjoy your stay!<br />"
   port=<your_port>
-  serverpassword=
+  serverpassword=changeThisPassword!
   allowping=true
   bonjour=false
+
 
 For more configuration options look at the documentation_ site.
 
@@ -131,7 +120,7 @@ Create ``~/etc/services.d/mumble.ini`` with the following content:
 .. code-block:: ini
 
   [program:mumble]
-  command=%(ENV_HOME)s/mumble/murmur.x86 -fg -ini %(ENV_HOME)s/mumble/murmur.ini
+  command=%(ENV_HOME)s/mumble/murmur.x86_64 -fg -ini %(ENV_HOME)s/mumble/murmur.ini
 
 .. include:: includes/supervisord.rst
 
@@ -141,7 +130,7 @@ If it's not in state ``RUNNING``, check your configuration.
 Finishing installation
 ======================
 
-To connect to your server, open mumble on your computer or phone and add a new server:
+To connect to your server, open a Mumble client on your computer or phone and add a new server:
 
 * Address: ``<username>.uber.space``
 * Port: your port
@@ -150,14 +139,15 @@ To connect to your server, open mumble on your computer or phone and add a new s
 Best practices
 ==============
 
-uberspace already provides letsencrypt certificates. To add them append these lines to your ``murmur.ini`` and adjust your username and the domain.
+Uberspace already provides letsencrypt certificates. To add them, append these lines to your ``murmur.ini`` and adjust your username and the domain.
 
 .. code-block:: ini
 
   sslCert=/home/isabell/etc/certificates/isabell.uber.space.crt
   sslKey=/home/isabell/etc/certificates/isabell.uber.space.key
 
-Last we have to restart the server before expiry. Edit your cron tab using the ``crontab -e``.
+The server needs to be restarted, before the certificates expire.
+Therefore, add the following line to your contrab using ``crontab -e``:
 
 ::
 
@@ -166,7 +156,7 @@ Last we have to restart the server before expiry. Edit your cron tab using the `
 Updates
 =======
 
-.. note:: Check the update feed_ regularly to stay informed about the newest version.
+.. note:: `Subscribe to the releases <https://github.com/franok/mumble-build-container/releases>`_ to stay informed about the latest versions (e.g. by using :lab:`uu-notify <guide_uu-notify>`).
 
 If there is a new version repeat the Installation_ and execute:
 
@@ -179,15 +169,14 @@ If there is a new version repeat the Installation_ and execute:
 
 .. _Teamspeak: https://www.teamspeak.com/
 .. _Mumble: https://mumble.info/
-.. _Github: https://github.com/mumble-voip/mumble
-.. _MySQL: https://manual.uberspace.de/en/database-mysql.html
+.. _GitHub: https://github.com/franok/mumble-build-container
+.. _`GitHub release section`: https://github.com/franok/mumble-build-container/releases
 .. _supervisord: https://manual.uberspace.de/en/daemons-supervisord.html
-.. _feed: https://github.com/mumble-voip/mumble/releases
 .. _documentation: https://wiki.mumble.info/wiki/Murmur.ini
 
 
 ----
 
-Tested with Murmur v1.3.4, Uberspace v7.11.1
+Tested with self-compiled static Murmur v1.4.287, Uberspace v7.14.0
 
 .. author_list::

--- a/source/guide_mumble.rst
+++ b/source/guide_mumble.rst
@@ -1,4 +1,5 @@
-.. author:: stunkymonkey <http://stunkymonkey.de>, updated by franok <https://franok.de>
+.. author:: stunkymonkey <http://stunkymonkey.de>
+.. author:: franok <https://franok.de>
 .. highlight:: console
 
 .. tag:: lang-cpp

--- a/source/guide_mumble.rst
+++ b/source/guide_mumble.rst
@@ -84,6 +84,14 @@ Extract the zip file:
     inflating: murmur.ini
     inflating: murmur.x86_64
 
+
+
+Make the binary executable:
+
+
+.. code-block:: console
+
+  [isabell@stardust mumble]$ chmod +x murmur.x86_64
   [isabell@stardust mumble]$
 
 

--- a/source/guide_mumble.rst
+++ b/source/guide_mumble.rst
@@ -158,7 +158,17 @@ Updates
 
 .. note:: `Subscribe to the releases <https://github.com/franok/mumble-build-container/releases>`_ to stay informed about the latest versions (e.g. by using :lab:`uu-notify <guide_uu-notify>`).
 
-If there is a new version repeat the Installation_ and execute:
+If there is a new version, backup your mumble folder:
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ cp -rp mumble/ mumble-backup/
+  [isabell@stardust ~]$ ls -la | grep -i mumble-backup
+  drwxrwxr-x.   4 isabell  isabell    167 Dec  6 20:16 mumble-backup
+  [isabell@stardust ~]$ 
+
+
+Repeat the Installation_ steps, and finally execute:
 
 .. code-block:: console
 
@@ -166,6 +176,14 @@ If there is a new version repeat the Installation_ and execute:
   mumble: stopped
   mumble: started
   [isabell@stardust ~]$
+
+Connect to your server and if everything works fine, delete the backup:
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ rm -rf mumble-backup/
+  [isabell@stardust ~]$
+
 
 .. _Teamspeak: https://www.teamspeak.com/
 .. _Mumble: https://mumble.info/

--- a/source/guide_mumble.rst
+++ b/source/guide_mumble.rst
@@ -79,14 +79,10 @@ Extract the zip file:
 
 .. code-block:: console
 
-  [isabell@stardust mumble]$ unzip -l murmur-static.zip
+  [isabell@stardust mumble]$ unzip murmur-static.zip
   Archive:  murmur-static.zip
-    Length      Date    Time    Name
-  ---------  ---------- -----   ----
-      18545  2022-12-06 21:27   murmur.ini
-   18621464  2022-12-06 21:28   murmur.x86_64
-  ---------                     -------
-  18640009                     2 files
+    inflating: murmur.ini
+    inflating: murmur.x86_64
 
   [isabell@stardust mumble]$
 

